Python client contributing (#1228)
G-Research/armada: client/python/CONTRIBUTING.md

How to develop on this python client project 
=
<hr />

Armada Project welcomes contributions from the opensource community.

> ### This client is written primarily targeting supported versions PYTHON 3 on Linux, Mac, and WSL on Windows.

Reviewer: ```suggestion
This client is written primarily targeting supported versions of python 3 on Linux, Mac, and WSL on Windows.
```
Just making this no longer a header, and cleaning up the wording

How to develop on this python client project 
=
<hr />

Armada Project welcomes contributions from the opensource community.

> ### Note that you need PYTHON 3+ to get this project working and this project does not work on native windows OS.

Reviewer: ```suggestion
This client is written primarily targeting supported versions of python 3 on Linux, Mac OS, and WSL on Windows. 
```

I suggest we make this more declarative, we can't list *all* the things we don't support so lets just list the things we do support.

> ## Setting up your own fork of this repository on linux, Mac, and WSL OS.

Reviewer: ```suggestion
## Setting up your own fork of this repository on Linux, Mac, or WSL OS.
```

<hr />

1) On github interface click on the `Fork` button.

> 2) Clone your forked of armada repository on your personal git repository by running git clone `github.com:Your_Git_Username/armada.git`.

Reviewer: ```suggestion
2) Clone your forked of armada repository on your personal git repository by running `git clone github.com:Your_GitHub_Username/armada.git`.
```

3) Enter the directory `cd armada`

> 4) Add upstream repository [Armada Repo](https://github.com/G-Research/armada.git)

Reviewer: ```suggestion
4) Add upstream repository `git remote add upstream https://github.com/G-Research/armada.git`
```

> ## Install Pyenv/Python and Poetry

Reviewer: We're about to merge a change in how the client is built, which invalidates some of this thru no fault of your own.

```suggestion
## Install Pyenv/Python
```

<hr />

2) Install [Pyenv](https://github.com/pyenv/pyenv) by following the instructions [here](https://realpython.com/intro-to-pyenv/).

> 3) Install [poetry](https://python-poetry.org/) on your local machine by following the instructions on [python poetry documentation](https://python-poetry.org/docs/).

Reviewer: This will no longer be required

## Setting up your own environment
<hr />

> 1) Run `make python` then `cd client/python` and run `poetry install` to pull all necessary dependencies.

Reviewer: ```suggestion
1) Run `make python` then `cd client/python` and run `pip install .` to install the client and all dependencies. You need to re-run the pip install command as you make changes.
```

> ## Run test to ensure everything is working

Reviewer: We should encourage people to run tests via tox.

`tox`, from the root of client/python, should be all people need to run to get unit tests and formatting run.
